drivers/python/rethinkdb/net.py

```diff
@@ -401,12 +403,14 @@ def _get_ssl_context(self, ca_certs):
 
 
 class ConnectionInstance(object):
-    def __init__(self, parent):
+    def __init__(self, parent, json_encoder=None, json_decoder=None):
```

Comment:
The `json_encoder` and `json_decoder` code appears to be repeated verbatim for each type of connection.  This code should probably be put straight into the `Connection` class, which is used as the base class for the different types - `DefaultConnection`, `net_twisted.Connection`, `net_tornado.Connection`, and `net_asyncio.Connection`.

Since this code is common to all the connection types (and doesn't really rely on any ConnectionInstance variables), `Connection` should just pull `json_encoder` and `json_decorder` out of the kwargs, and provide the `_get_json_decoder` and `_get_json_encoder` functions that the child classes will inherit.